src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -146,42 +146,36 @@ void GcsActorManager::HandleGetActorInfo(const rpc::GetActorInfoRequest &request
   RAY_LOG(DEBUG) << "Getting actor info"
                  << ", job id = " << actor_id.JobId() << ", actor id = " << actor_id;
 
-  auto on_done = [actor_id, reply, send_reply_callback](
-                     const Status &status,
-                     const boost::optional<ActorTableData> &result) {
-    if (result) {
-      reply->mutable_actor_table_data()->CopyFrom(*result);
+  const auto &registered_actor_iter = registered_actors_.find(actor_id);
+  if (registered_actor_iter != registered_actors_.end()) {
+    reply->mutable_actor_table_data()->CopyFrom(
+        registered_actor_iter->second->GetActorTableData());
+  } else {
+    const auto &destroyed_actor_iter = destroyed_actors_.find(actor_id);
```

Comment:
Isn't it hurting long-running clusters? Are we going to store all the past actor data in-memory? We can probably pull it from Redis when the actor is already dead?

Reply:
Yeah, I will submit a PR and delete the offline job data regularly according to the job dimension. So memory doesn't always inflate. Now if the job is offline, the non-detached actor will not be deleted from redis. I will modify it together.

Reply:
@rkooo567 I want to remove the spec of dead actor, because it takes up a lot of memory.

Reply:
Maybe we can just have a cache and evict a random entry? Upon cache miss, we can probably just query from Redis.

Reply:
Or another idea is to always query Redis for destroyed actors. I think `ray.actors()` is not commonly used, so it might be fine. We can also reduce the payload by only returning actors that are associated with the job (+ detached actors)

Reply:
@ffbin Are you going to implement these things in the current PR, or will you have a followup?

Reply:
I want to implement these things in next PR.

Reply:
How about have a cache and evict a random entry when exceeding? haha

Reply:
I will also test whether the performance is improved after the spec was removed. If it's faster, maybe we can query it directly from redis.